source/screenExplorer.py

```diff
@@ -1,8 +1,68 @@
+#screenExplorer.py
+#A part of NonVisual Desktop Access (NVDA)
+#Copyright (C) 2012-2018 NV Access Limited, Joseph Lee
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+
+"""Module which implements various utilities for screen exploration.
+This includes touchscreen movement as well as handling coordinate announcement for mouse and objects.
+"""
+
 import api
 import tones
 import controlTypes
 import textInfos
 import speech
+import screenBitmap
+import config
+import mouseHandler
+import wx
+
+def playLocationCoordinates(x, y, screenWidth, screenHeight, screenMinPos=None, detectBrightness=True,blurFactor=0):
+	"""Plays a tone indicating the XY-coordinate for the given location. This works for keyboard-based object navigation/focus movement, mouse movement and during touch hover gesture.  
+	@param obj: the coordinate for the location and the screen resolution (typically desktopObject.location)
+	@type obj: int
+	"""
+	# Have a fake wx.Point in case min pos is not defined on multiple monitors.
+	if screenMinPos is None:
+		screenMinPos = wx.Point()
+	# Do not play any tone if offscreen.
+	if 0 <= x <= screenWidth or 0 <= y <= screenHeight:
+		minPitch=config.conf['mouse']['audioCoordinates_minPitch']
+		maxPitch=config.conf['mouse']['audioCoordinates_maxPitch']
+		curPitch=minPitch+((maxPitch-minPitch)*((screenHeight-y)/float(screenHeight)))
+		if detectBrightness:
+			startX=min(max(x-blurFactor,0),screenWidth)+screenMinPos.x
+			startY=min(max(y-blurFactor,0),screenHeight)+screenMinPos.y
+			width=min(blurFactor+1,screenWidth)
+			height=min(blurFactor+1,screenHeight)
+			grey=screenBitmap.rgbPixelBrightness(mouseHandler.scrBmpObj.captureImage( startX, startY, width, height)[0][0])

```

Comment:
Could you move the scrBmpObj to screenExplorer as well? It currently makes no sense within the scope of mouseHandler.

Reply:
Hi,

Hmmm, the object itself is initialized and nullified when mouse handler starts and ends, respectively. We need to gather more feedback regarding this, as it could break features if this is moved without careful thinking. Thanks.

Reply:
There are no other references to `scrBmpObj` outside of `mouseHandler.py`, you can confirm with `git grep -i scrBmpObj`.

Are you worried that addons are using it? My preference would be to understand its usage, and what the intention was behind initialising it this way, with the goal of moving it out of `mouseHandler`